### ISSUE TITLE
Soft input keyboard stays open?

### ISSUE THREAD

**Issue by Bill871** [afollestad/material-dialogs#978]
I think that even after #971 fix, this is still broken in 0.8.5.6. Or at least the behavior changed and in my opinion the SHOW_IMPLICIT worked better:

- when you have an dialog with input text, the keyboard opens automatically, which is alright
- but when you tap an button to open another dialog over this one that does NOT have an input text, the keyboard now stays open (which in my opinion is wrong), in version 0.8.5.4 this worked correctly

**Comment by afollestad** [comment]
It works fine for me.

Feel free to make a pull request and test it. I don't have time to test/fix things right now.

**Comment by Bill871** [comment]
From a little testing, it looks like that the opening of the 2nd dialog and closing of the soft keyboard is dependent on the fact whether the 1st dialog is using custom view or not.

1) when 1st dialog is built using .customView() with input inside of it - opening an 2nd dialog without input closes the keyboard (correct)

2) but when 1st dialog is built using a builder with .input("", "", ...) - opening an 2nd dialog without input does NOT close the keyboard (wrong)

**Comment by afollestad** [comment]
`MaterialDialog` will always dismiss the soft keyboard when it is cancelled *or* dismissed. That includes through the `dismiss()` method, `cancel()` method, or when the user taps outside of the dialog.

**Comment by Bill871** [comment]
Well, true. But in my opinion the 2) should work (as it worked in 0.8.5.4). Never mind, I will revert back to 0.8.5.4 where it works correctly, too bad that I do have to revert also to 23.1.1 support libraries. Thanks for your work, though.

**Comment by Spoetnic** [comment]
I'm not sure if I should make a new issue, since I think its the same as this one.
(version 0.8.5.6, Marshmallow)
My softInputKeyboard wouldnt close after dismissing a dialog, so I wanted to do it myself in an onCancel or onDismissed listener. These do not seem to be called though.
I even tried Builder.build().setOnCancelListener(...), after the call on the builder yielded no result.
The decompiled sourcecode that Android Studio provides shows that these 2 listeners (in the builder) are only set, but never used.

Any thoughts on this.?

**Comment by FrancescoCapari** [comment]
In DialogUtils#showKeyboard() the second parameter into the showSoftInput() method should be InputMethodManager.SHOW_IMPLICIT instead of InputMethodManager.SHOW_FORCED. That would fix all these issues. FORCED should be used if any only if user e.g. pressed the key to display the keyboard. Since this method is called onShow() of any dialog, this is obviously not the case.

**Comment by afollestad** [comment]
https://github.com/afollestad/material-dialogs/releases/tag/0.8.6.1

**Comment by othreecodes** [comment]
Hey, Im getting this same Issue while using Material Dialog in a fragment
